src/dtproject/types.py

```diff
@@ -126,9 +126,11 @@ class ContainerConfiguration(dict):
 class LayerContainers(DictLayer[ContainerConfiguration]):
     ITEM_CLASS = ContainerConfiguration
 
-
+@dataclasses.dataclass
 class DevContainerConfiguration(dict):
-    pass
+    docker_compose_file: str
+    service: str
+    workspace_folder: str
```

Comment:
This would be similar to the `LayerContainers` and `ContainerConfiguration` classes.

e.g.,
```Python

@dataclasses.dataclass
class DevContainerConfiguration:
    configuration: str
    <any devcontainer config properties that are not provided by a docker-compose file>


@dataclasses.dataclass
class LayerDevContainers(DictLayer[DevContainerConfiguration]):
    ITEM_CLASS = DevContainerConfiguration

```